code_detection/base.py

```diff
@@ -29,7 +29,7 @@ def text(self) -> str:
         # there is little to no purpose for blank lines at the start or end of a section
         lines = list(self.lines)
         for end in (0, -1):
-            while lines and not lines[end] or lines[end].isspace():
+            while not lines[end] if lines else lines[end].isspace():
```

Comment:
This is equivalent, but I'm noticing that the original code is actually wrong lol - it should be
```suggestion
            while lines and (not lines[end] or lines[end].isspace()):
```

because both of the `or` conditions require that `lines` is not empty